<launch>
<!-- complete.launch -->
<!-- actually it's missing mapping-->
	<group>
		<node name="RxTx" pkg="rxtx" type="RxTx" output="screen">
		  <param name="SerialPort" value="/dev/ttyUSB0" />
		  <param name="BaudRate" value="38400" />
		</node>
		<node name="HoverboardLL" pkg="hoverboard" type="HoverboardLL"  output="screen" />
		<node name="Hovercraft" pkg="hovercraft" type="HoverCraft"  output="screen" />
	</group>
	<group>
		<node respawn="true" pkg="joy"
			type="joy_node" name="xbox_joy" >
		   <param name="dev" type="string" value="/dev/input/js0" />
		   <param name="deadzone" value="0.12" />
		</node>
		<node name="joystickControls" pkg="lab2" type="joystick.py" output="screen">
			<param name="debug" type="string" value="0" />
		</node>
		<node name="angleIntegrator" pkg="lab2" type="angleIntegrator.py" output="screen">
			<param name="debug" type="string" value="0" />
		</node>
	</group>
	<group>
		<node name="RangeIntegrator" pkg="lab2" type="rangeFinderIntegrator.py" output="screen">
            <param name="debug" type="string" value="0"/>
		</node>
		<node name="reactiveControl" pkg="lab2" type="reactiveControl.py"  output="screen" >
            <param name="debug" type="string" value="0"/>
            <param name="P" type="string" value="0.5"/>
            <param name="D" type="string" value="0"/>
        </node>
    </group> 
    <group>
        <node name="Mapping" pkg="lab2" type="mapping.py" output="screen">
            <param name="debug" type="string" value="1"/>
            <param name="rotationalUnit" type="string" value="100"/>
        </node>
    </group>
	<node name="triangle" pkg="lab2" type="triangle.py" output="screen" >
		<param name="debug" type="string" value="0"/>
	</node>
    <node name="Switch" pkg="lab2" type="switch.py" output="screen" >
        <param name="debug" type="string" value="1"/>
    </node>    
	<node name="Arbitrator" pkg="lab2" type="arbitrator.py" output="screen" >
		<param name="debug" type="string" value="0" />
	</node>
	
	<group>
		<node name="AngularPositionPID" pkg="lab2" type="angularPositionPID.py" output="screen" >
			<param name="P" type="string" value="1"/>
			<param name="D" type="string" value="0.2"/>
			<param name="debug" type="string" value="0" />
		</node>
		<node name="AngularVelocityPID" pkg="lab2" type="angularVelocityPID.py" output="screen" >
			<param name="targetRate" type="string" value="10"/>
			<param name="P" type="string" value=".5"/>
			<param name="D" type="string" value="0.3"/>
			<param name="debug" type="string" value="0" />
		</node>
	</group>
	<node name="thrusterMapping" pkg="lab2" type="thrusterMapping.py" output="screen" >
		<param name="debug" type="string" value="0" />
		<param name="liftPower" type="string" value="0.3" />
	</node>

</launch>

